src/scxa2cxg.py

```diff
@@ -0,0 +1,114 @@
+import json
+import os
+from ftplib import FTP
+
+import anndata as ad
+import httpx
+import pandas as pd
+
+BASE_URL = "https://ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/sc_experiments/{study}/"
+H5AD_EXT_FILE = ".project.h5ad"
+METADATA_EXT_FILE = ".cell_metadata.tsv"
+
+
+def download_files(study: str):
+    """
+    Download H5AD and metadata files for a study
+    """
+    download_dir = os.path.join("downloads")
+    study_dir = os.path.join(download_dir, study)
+    h5ad_path = f"{study_dir}/{study}{H5AD_EXT_FILE}"
+
+    if not os.path.isdir(download_dir):
+        os.mkdir(download_dir)
+
+    if not os.path.isdir(study_dir):
+        os.mkdir(study_dir)
+
+    if not os.path.exists(h5ad_path):
+        res = httpx.get(f"{BASE_URL.format(study=study)}{study}{H5AD_EXT_FILE}")
+        with open(h5ad_path, 'wb') as f:
+            f.write(res.content)
+
+
+def convert(study: str):
+    """
+    Change column names to be conform to CxG schema v4.0.0
+    """
+    ann_data = ad.read_h5ad(
+        f"downloads/{study}/{study}{H5AD_EXT_FILE}",
+        backed="r+"
+    )
+    new_obs = pd.DataFrame(
+        data=ann_data.obs.copy(),
+        index=ann_data.obs.index.copy()
+    )
+    obs_cols = ann_data.obs_keys()
+
+    new_obs.rename(columns={
+        "developmental_stage": "development_stage",
+        "developmental_stage_ontology": "developmental_stage_ontology_term_id",
+        "inferred_sex": "sex",
+        "inferred_sex_ontology": "sex_ontology_term_id",
+        "sex_ontology": "sex_ontology_term_id",
+        "cell_type_ontology": "cell_type_ontology_term_id", 
+        "inferred_cell_type_-_ontology_labels": "cell_type",
+        "inferred_cell_type_-_ontology_labels_ontology": "cell_type_ontology_term_id",
+        "authors_cell_type_-_ontology_labels": "cell_type",
+        "authors_cell_type_-_ontology_labels_ontology": "cell_type_ontology_term_id",
```

Comment:
`cell_type` and `cell_type_ontology_term_id` are duplicated. Is this correct?

Reply:
I found in some files they use different names for the cell type. I've got the obs columns for each experiment in here https://github.com/EBISPOT/scxa_kg/blob/3b76ab7c605c6fda263d00421f99494b16a81007/obs_columns_by_experiment.json